actors/builtin/miner/miner_actor.go

```diff
@@ -1465,9 +1630,9 @@ func (a Actor) CompactPartitions(rt Runtime, params *CompactPartitionsParams) *a
 		info := getMinerInfo(rt, &st)
 		rt.ValidateImmediateCallerIs(append(info.ControlAddresses, info.Owner, info.Worker)...)
 
-		if !deadlineIsMutable(st.ProvingPeriodStart, params.Deadline, rt.CurrEpoch()) {
+		if !deadlineCanCompact(st.ProvingPeriodStart, params.Deadline, rt.CurrEpoch()) {
```

Comment:
From my discussion at the top level, I'd like to unify these again if it's reasonable, with a longer immutability period.

Reply:
Unfortunately, we can't do this easily. That would prevent us from assigning sectors to partitions in that window which would mean opening even _more_ partitions, which would be very painful for small miners.

* Adding a sector to a partition doesn't affect window PoSt verification/fees/faults.
* Compaction does.